dota2/features/lobby.py

```diff
@@ -0,0 +1,330 @@
+
+        self.socache.on(('new', ESOType.CSODOTALobby), self.__handle_lobby_new)
+        self.socache.on(('updated', ESOType.CSODOTALobby), self.__handle_lobby_changed)
+        self.socache.on(('removed', ESOType.CSODOTALobby), self.__handle_lobby_removed)
+
+    def __lobby_cleanup(self):
+        self.lobby = None
+
+    def __handle_lobby_invite(self, message):
+        self.emit(self.EVENT_LOBBY_INVITE, message)
+
+    def __handle_lobby_invite_removed(self, message):
+        self.emit(self.EVENT_LOBBY_INVITE_REMOVED, message)
+
+    def __handle_lobby_new(self, message):
+        self.lobby = message
+        self.emit(self.EVENT_LOBBY_NEW, message)
+
+    def __handle_lobby_changed(self, message):
+        self.lobby = message
+        self.emit(self.EVENT_LOBBY_CHANGED, message)
+
+    def __handle_lobby_removed(self, message):
+        self.lobby = None
+        self.emit(self.EVENT_LOBBY_REMOVED, message)
+
+    def create_practice_lobby(self, password="", options=None):
+        """
+        Sends a message to the Game Coordinator requesting to create a lobby.
+
+        :param password: password of lobby
+        :type password: :class:`str`
+        :param options: options to setup the lobby with
+        :type options: :class:`dict`
+        """
+        return self.create_tournament_lobby(password=password, options=options)
+
+    def create_tournament_lobby(self, password="", tournament_game_id=None, tournament_id=0, options=None):
+        """
+        Sends a message to the Game Coordinator requesting to create a tournament lobby.
+
+        :param password: password of lobby
+        :type password: :class:`str`
+        :param tournament_game_id: tournament game id
+        :type tournament_game_id: :class:`int`
+        :param tournament_id: tournament id
+        :type tournament_id: :class:`int`
+        :param options: options to setup the lobby with
+        :type options: :class:`dict`
+        """
+        options = {} if options is None else options
+        options["pass_key"] = password
+
+        command = {
+            "lobby_details": options,
+            "pass_key": password
+        }
+        if tournament_game_id is not None:
+            command.update({
+                "tournament_game": True,
+                "tournament_game_id": tournament_game_id,
+                "tournament_id": tournament_id
+            })
+
+        if self.verbose_debug:
+            self._LOG.debug("Sending match CMsgPracticeLobbyCreate request")
+
+        self.send(EDOTAGCMsg.EMsgGCPracticeLobbyCreate, command)
+
+    def config_practice_lobby(self, lobby_id, options):
+        """
+        Change settings of the selected lobby.
+
+        :param lobby_id: target lobby
+        :type lobby_id: :class:`int`
+        :param options: options to change in the lobby
+        :type options: :class:`dict`
+        """
+        options = {} if options is None else options
+        options['lobby_id'] = lobby_id
```

Comment:
You are right, I started from the node interfaces which are using `lobby_id` because they don't have the `lobby` argument.